Building typings for a project that imports Three.JS fails

I'm trying to build an amalgamated and flattened typings file with Three.JS as a dependency. I have narrowed down the problem to this minimal example. I would be very thankful for your help with this, the goal is to have a typings file for my own code and let it rely on typings from NPM for dependencies like Three.JS. I would rather not want to inline all the typings.

app.ts:
```ts
import {Vector3} from "three";

export function test(): Vector3 {
  return new Vector3();
}
```
package.json:
```json
{
  "name": "dts-bundle-generator-bugreport",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "build": "npx tsc",
    "typings": "npx dts-bundle-generator -o app.d.ts app.ts"
  },
  "dependencies": {
    "three": "^0.93.0"
  },
  "devDependencies": {
    "dts-bundle-generator": "^1.2.2",
    "typescript": "^2.9.1",
    "@types/three": "^0.92.3"
  }
}
```
tsconfig.json:
```json
{
  "compilerOptions": {
    "target": "es5",
    "declaration": true,
    "lib": ["es2015", "dom"]
  }
}
```
Run:
```bash
npm install
npm run typings
```
Output:
```
Library "three" will be added via reference directive
Library "webvr-api" will be added via reference directive
Writing generated file to app.d.ts...
Checking the generated file...
app.d.ts(4,33): error TS2304: Cannot find name 'Vector3'.
Error: Compiled with errors
```
app.d.ts looks like:
```ts
/// <reference types="three" />
/// <reference types="webvr-api" />

export declare function test(): Vector3;
```

Question / bug (?): Why is the library added via reference, the documentation states "By default all libraries will be imported (except inlined)"? And why does the check fail? I'm not too familiar with the reference syntax, but I doubt that it makes the type `Vector3` available, so the check is probably right.

Next attempt: set `--external-imports` to a package name from `node_modules` to force the use of import:
```diff
-    "typings": "npx dts-bundle-generator -o app.d.ts app.ts"
+    "typings": "npx dts-bundle-generator -o app.d.ts --external-imports @types/three -- app.ts"
```

Exactly the same output as before. Next try:

```diff
-    "typings": "npx dts-bundle-generator -o app.d.ts app.ts"
+    "typings": "npx dts-bundle-generator -o app.d.ts --external-imports three -- app.ts"
```

Output:
```
Adding import with name "Vector3" for library "@types/three"
Library "webvr-api" will be added via reference directive
Writing generated file to app.d.ts...
Checking the generated file...
app.d.ts(3,25): error TS6137: Cannot import type declaration files. Consider importing 'three' instead of '@types/three'.
Error: Compiled with errors
```

app.d.ts:
```ts
/// <reference types="webvr-api" />

import { Vector3 } from '@types/three';

export declare function test(): Vector3;
```

Almost, it should indeed be `from 'three'` and not `from '@types/three'`.

--- a/README.md
+++ b/README.md
@@ -73,7 +73,7 @@ Options:
                            Used types will be imported by "import { First,
                            Second } from 'library-name';".
                            By default all libraries will be imported (except
-                           inlined)                                      [array]
+                           inlined and libraries from @types)            [array]
   --external-types         Array of the package names from @types to import
                            typings from it via triple-slash reference directive.
                            By default all packages are allowed and will be used

--- a/src/bin/dts-bundle-generator.ts
+++ b/src/bin/dts-bundle-generator.ts
@@ -71,7 +71,7 @@ const args = yargs
 		type: 'array',
 		description: 'Array of the package names from node_modules to import typings from it.\n' +
 			'Used types will be imported by "import { First, Second } from \'library-name\';".\n' +
-			'By default all libraries will be imported (except inlined)',
+			'By default all libraries will be imported (except inlined and libraries from @types)',
 		coerce: toStringsArray,
 	})
 	.option('external-types', {

--- a/src/module-info.ts
+++ b/src/module-info.ts
@@ -89,7 +89,7 @@ function getModuleInfoImpl(currentFilePath: string, originalFileName: string, cr
 	}
 
 	if (shouldLibraryBeImported(npmLibraryName, typesLibraryName, criteria.importedLibraries)) {
-		return { type: ModuleType.ShouldBeImported, fileName: originalFileName, libraryName: npmLibraryName, isExternal: true };
+		return { type: ModuleType.ShouldBeImported, fileName: originalFileName, libraryName: typesLibraryName || npmLibraryName, isExternal: true };
 	}
 
 	if (typesLibraryName !== null && isLibraryAllowed(typesLibraryName, criteria.allowedTypesLibraries)) {

--- a/tests/test-cases/import-from-types/config.ts
+++ b/tests/test-cases/import-from-types/config.ts
@@ -2,7 +2,7 @@ import { TestCaseConfig } from '../test-case-config';
 
 const config: TestCaseConfig = {
 	generatorOptions: {
-		importedLibraries: ['fs'],
+		importedLibraries: ['fs', 'fake-types-lib'],
 	},
 };
 

--- a/tests/test-cases/import-from-types/input.ts
+++ b/tests/test-cases/import-from-types/input.ts
@@ -1,5 +1,7 @@
+import { InterfaceFromTypesPackage } from 'fake-types-lib';
 import { Stats } from 'fs';
 
 export interface InterfaceName {
 	prop: Stats;
+	field: InterfaceFromTypesPackage;
 }

--- a/tests/test-cases/import-from-types/output.d.ts
+++ b/tests/test-cases/import-from-types/output.d.ts
@@ -1,5 +1,7 @@
+import { InterfaceFromTypesPackage } from 'fake-types-lib';
 import { Stats } from 'fs';
 
 export interface InterfaceName {
 	prop: Stats;
+	field: InterfaceFromTypesPackage;
 }

